src/App.js

```diff
@@ -1,17 +1,110 @@
 import React from 'react';
 import './App.css';
+import ClassNames from 'classnames';
+import { TodoList } from './componenets/TodoList';
 
-import users from './api/users';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-function App() {
-  return (
+class App extends React.Component {
+  state = {
+    todos: todosFromServer.map(todo => ({
+      ...todo,
+      user: usersFromServer.find(user => user.id === todo.userId),
+    })),
+    isHidden: true,
+    errorMessage: '',
+  }
+
+  timeToShowError = false;
+
+  validation = (event) => {
+    const form = event.target.closest('#formAdder');
+
+    if (!form.title.value && this.timeToShowError) {
+      this.setState({
+        isHidden: false,
+        errorMessage: 'enter title',
+      });
+
+      return false;
+    }
+
+    if (!form.userSelect.value && this.timeToShowError) {
+      this.setState({
+        isHidden: false,
+        errorMessage: 'chose user',
+      });
+
+      return false;
+    }
+
+    this.setState({
+      isHidden: true,
+      errorMessage: '',
+    });
+
+    return true;
+  }
+
+  Add = (event) => {
```

Comment:
why uppercase?

Reply:
because 
![image](https://user-images.githubusercontent.com/41700032/92452108-47e17d00-f1c6-11ea-966d-a0ad0cdb0884.png)

Reply:
this is the name for the button, not the method